File: app/src/androidTest/java/com/dialogapp/dialog/db/DbTest.java
package com.dialogapp.dialog.db;

import androidx.room.Room;
import androidx.test.InstrumentationRegistry;

import org.junit.After;
import org.junit.Before;

public abstract class DbTest {
    protected MicroBlogDb db;

    @Before
    public void setUp() {
        db = Room.inMemoryDatabaseBuilder(InstrumentationRegistry.getContext(), MicroBlogDb.class)
                .build();
    }

    @After
    public void tearDown() throws Exception {
        db.close();
    }
}

File: app/src/androidTest/java/com/dialogapp/dialog/db/DbTest.kt
package com.dialogapp.dialog.db

import androidx.arch.core.executor.testing.CountingTaskExecutorRule
import androidx.room.Room
import androidx.test.InstrumentationRegistry

import org.junit.After
import org.junit.Before
import org.junit.Rule
import java.util.concurrent.TimeUnit

abstract class DbTest {
    @Rule
    @JvmField
    val countingTaskExecutorRule = CountingTaskExecutorRule()
    private lateinit var _db: MicroBlogDb
    val db: MicroBlogDb
        get() = _db

    @Before
    fun initDb() {
        _db = Room.inMemoryDatabaseBuilder(
                InstrumentationRegistry.getContext(),
                MicroBlogDb::class.java
        ).build()
    }

    @After
    fun closeDb() {
        countingTaskExecutorRule.drainTasks(10, TimeUnit.SECONDS)
        _db.close()
    }
}
